system/security/LdapSecurity/ldapsecurity.cpp

```diff
@@ -213,27 +213,27 @@ ISecUser * CLdapSecUser::clone()
  *     CLdapSecResource                                   *
  **********************************************************/
 
-CLdapSecResource::CLdapSecResource(const char *name) : m_name(name), m_access(0), m_required_access(0)
+CLdapSecResource::CLdapSecResource(const char *name) : m_name(name), m_access(SecAccess_None), m_required_access(SecAccess_None)
 {
     m_resourcetype = RT_DEFAULT;
 }
 
-void CLdapSecResource::addAccess(int flags)
+void CLdapSecResource::addAccess(SecAccessFlags flags)
 {
-    m_access |= flags;
+    m_access = (SecAccessFlags)((int)m_access | (int)flags);
```

Comment:
this might be 100% legit, but the casting style seems vulnerable for unexpected behavior (in case of unexpected access flags)